File: app/src/main/java/md/fusionworks/lifehack/data/api/model/sales/SaleCategoryDataMapper.java
package md.fusionworks.lifehack.data.api.model.sales;

import java.util.ArrayList;
import java.util.List;
import md.fusionworks.lifehack.ui.sales.model.SaleCategoryModel;

/**
 * Created by ungvas on 2/18/16.
 */
public class SaleCategoryDataMapper {

  public SaleCategoryModel transform(SaleCategory saleCategory) {
    return new SaleCategoryModel(saleCategory.id, saleCategory.nameRu, saleCategory.nameRo);
  }

  public List<SaleCategoryModel> transform(List<SaleCategory> saleCategoryList) {
    List<SaleCategoryModel> saleCategoryModelList = new ArrayList<>(saleCategoryList.size());
    SaleCategoryModel saleCategoryModel;
    for (SaleCategory saleCategory : saleCategoryList) {
      saleCategoryModel = transform(saleCategory);
      if (saleCategoryModel != null) {
        saleCategoryModelList.add(saleCategoryModel);
      }
    }
    return saleCategoryModelList;
  }
}

File: app/src/main/java/md/fusionworks/lifehack/data/api/model/sales/SaleCategoryDataMapper.kt
package md.fusionworks.lifehack.data.api.model.sales

import md.fusionworks.lifehack.ui.sales.model.SaleCategoryModel

/**
 * Created by ungvas on 2/18/16.
 */
class SaleCategoryDataMapper {

  fun transform(saleCategory: SaleCategory): SaleCategoryModel {
    return SaleCategoryModel(saleCategory.id, saleCategory.nameRu, saleCategory.nameRo)
  }

  fun transform(saleCategoryList: List<SaleCategory>): List<SaleCategoryModel> {
    val saleCategoryModelList = arrayListOf<SaleCategoryModel>();
    var saleCategoryModel: SaleCategoryModel?
    for (saleCategory in saleCategoryList) {
      saleCategoryModel = transform(saleCategory)
      if (saleCategoryModel != null) {
        saleCategoryModelList.add(saleCategoryModel)
      }
    }
    return saleCategoryModelList
  }
}
